week-01-introduction.tex

```diff
@@ -130,31 +135,36 @@ \subsection{The \texttt{Math} library}
 
-Soon we will learn how to creating functions of our own that will be indistinguishable from built-in functions. By creating new functions of your own, you can extend the JavaScript language adding a new function to do exactly what you would like it to.
+Soon we will learn how to create functions of our own that will be indistinguishable from built-in functions. By creating new functions of your own, you can extend the JavaScript language adding a new function to do whatever you would like it to.
 
 \section{Four Friends of Programming}
 
-There are lots of things to learn about engineering. New technologies are built every single day. And it is difficult to keep up. The shear number of things to know can be overwhelming. But fear not! There actually aren't that many different things you can do with a programming language. In fact, there are only four.\marginnote{The fact that there are only four things you can do is something of a theoretical marvel. It is related to \emph{Church-Turing thesis} and lies as the heart of the foundations of computing. Lucky for us, it means we need to learn only four concepts. Actually the fourth one, organization and reuse, is merely a convenience for us humans.} Any new technology or technique you run across is a cousin to one or more of the four friends you will make in this chapter. The four things you can do in a programming language are:
+There are lots of things to learn about engineering. New technologies are built every single day. And it is difficult to keep up. The sheer number of things to know can be overwhelming. But fear not! There actually aren't that many different things you can do with a programming language. In fact, there are only four.\marginnote{The fact that there are only four things you can do is something of a theoretical marvel. It is related to \emph{Church-Turing thesis} and lies as the heart of the foundations of computing. Lucky for us, it means we need to learn only four concepts. Actually the fourth one, organization and reuse, is merely a convenience for us humans.} Any new technology or technique you run across is a cousin to one or more of the four friends you will make in this chapter. The four things you can do in a programming language are:
+% @SDI: Screw convenience. Introduce your students to Turing Tarpits first thing! (No, don't.)
 
 \begin{enumerate}
   \item \textbf{Store data and retrieve it later to use.}
   Your friend for the pillar of data storage and retrieval is a \emph{constant}. Other instances of this pillar include files, databases, networked resources, and web pages on the Internet. At their heart, however, they're all about stashing some information somewhere and fetching it later.
+  % @SDI: "Variables" aren't worth a mention? They're as much a workhorse as a constant is.
```

Comment:
I had a huge internal struggle when I was racing to write this yesterday. I almost called constants "constant variables". I still might. And I will call out variables.

Reply:
It's fun to see the errors you make when you rush. It's clear the "English generator" part of your brain works very differently than mine!